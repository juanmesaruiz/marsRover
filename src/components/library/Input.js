import React from 'react';
import styled from 'styled-components';

const Input = React.forwardRef((props, ref) => <input {...props} ref={ref} />)

const getStylesBasedOnType = type => {
  switch (type) {
    case 'submit':
    case 'reset':
      return 'cursor: pointer;';
    default:
      return '';
  }
};

const StyledInput = styled(Input)`
  ${({ type }) => getStylesBasedOnType(type)}
`;

export default StyledInput;
