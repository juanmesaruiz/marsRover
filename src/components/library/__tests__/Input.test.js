import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import Input from '../Input';

describe('Input test suite', () => {
  const getComponent = props => <Input {...props} />;

  const setup = props => render(getComponent(props));

  it('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(getComponent(), div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Should match snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('Should match snapshot if Input has type submit', () => {
    const { container } = setup({ type: 'submit' });
    expect(container).toMatchSnapshot();
  });

  it('Should match snapshot if Input has type reset', () => {
    const { container } = setup({ type: 'reset' });
    expect(container).toMatchSnapshot();
  });
});
