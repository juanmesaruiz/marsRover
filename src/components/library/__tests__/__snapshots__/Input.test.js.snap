// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input test suite Should match snapshot 1`] = `
<div>
  <input
    class=""
  />
</div>
`;

exports[`Input test suite Should match snapshot if Input has type reset 1`] = `
.c0 {
  cursor: pointer;
}

<div>
  <input
    class="c0"
    type="reset"
  />
</div>
`;

exports[`Input test suite Should match snapshot if Input has type submit 1`] = `
.c0 {
  cursor: pointer;
}

<div>
  <input
    class="c0"
    type="submit"
  />
</div>
`;
